File: FrontUI/Angular-WebUI/client-app/src/app/Invoice/model/invoice.ts
import * as moment from 'moment';
import { Client } from '../../client/model/client';
import { ISearchable } from '../../common/ISearchable';
import { InvoiceLine } from './InvoiceLine';
import { InvoiceStatus } from './InvoiceStatus';

export class Invoice implements ISearchable {

        constructor(id: number) {
                this.id = id;
        }

        id: number;
        invoiceId!: string;
        billingDate!: Date;
        dueDate: number =45;
        message: string = "";
        client!:Client;
        status!:InvoiceStatus;
        invoiceLines!:InvoiceLine[];
}

File: src/Domain/Entities/Invoice.cs
namespace ERP.Domain.Entities;

public class Invoice: BaseAuditableEntity
{
    public Invoice()
    {
        this.InvoiceLines = new List<InvoiceLine>();
        this.Status = InvoiceStatus.Created;
    }
    
    public int ClientId { get; set; }
    public Client Client { get; set; }

    public string Identifier { get; set; }
    
    public int DueDate { get; set; }
    
    public DateTime BillingDate { get; set; }
    
    public string? Message { get; set; }

    public InvoiceStatus Status { get; set; }
    public List<InvoiceLine> InvoiceLines { get; set; }
}

public enum InvoiceStatus
{
    Created,
    Sent,
    Paid,
    Late
}